devicetypes/smartthings/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,277 @@
+	(endpoint == 1) ? [name: "switch", value: cmd.value ? "on" : "off"] : [:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	if (cmd.commandClass == 0x6C && cmd.parameter.size >= 4) { // Supervision encapsulated Message		
+		cmd.parameter = cmd.parameter.drop(2)
+		cmd.commandClass = cmd.parameter[0]
+		cmd.command = cmd.parameter[1]
+		cmd.parameter = cmd.parameter.drop(2)
+	}
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x25: 1, 0x20: 1])
+	if (cmd.sourceEndPoint == 1) {
+		zwaveEvent(encapsulatedCommand, 1)
+	} else { // sourceEndPoint == 2
+		childDevices[0]?.handleZWave(encapsulatedCommand)
+		[:]
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
+	if (endpoint == null) log.debug("$device.displayName: $cmd")
+	else log.debug("$device.displayName: $cmd endpoint: $endpoint")
+}
+
+def on() {
+	// parent DTH controls endpoint 1
+	def endpointNumber = 1
+	delayBetween([
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinarySet(switchValue: 0xFF)),
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinaryGet())
+	])
+}
+
+def off() {
+	def endpointNumber = 1
+	delayBetween([
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinarySet(switchValue: 0x00)),
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinaryGet())
+	])
+}
+
+// PING is used by Device-Watch in attempt to reach the Device
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	[encap(1, zwave.switchBinaryV1.switchBinaryGet()), encap(2, zwave.switchBinaryV1.switchBinaryGet())]
+}
+
+// sendCommand is called by endpoint 2 child device handler
+def sendCommand(endpointDevice, commands) {
+	def endpointNumber = 2
+	def result
+	if (commands instanceof String) {
+		commands = commands.split(',') as List
+	}
+	result = commands.collect { encap(endpointNumber, it) }
+	sendHubCommand(result, 100)
+}
+
+def encap(endpointNumber, cmd) {
+	if (endpointNumber == 0) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	}
+	else if (cmd instanceof physicalgraph.zwave.Command) {
+		def cmdTemp = zwave.multiChannelV3.multiChannelCmdEncap(sourceEndPoint: 0x01, destinationEndPoint: endpointNumber).encapsulate(cmd)
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmdTemp).format()
+	} else {
+		cmd.format()
+	} 
+}
+
+private getParameterMap() {[[
+		name: "Input switch type (Default: Toggle Switch)", key: "inputSwitchType", type: "enum",
+		parameterNumber: 1, size: 1, defaultValue: 0,
+		values: [
+				0: "Toggle Switch",
+				1: "Momentary Switch",
+		],
+		description: "This item can select input switch type."
+	]
+]}
```

Comment:
since the device has only one preference, the code for handling them can be simplified considerably

Reply:
Can I keep this structure?
because our company plan develop new product,
it's base this device, will add more configuration parameter,
if I change it, I will change too more after new product release and upgrade DTH.